src/python/pants/backend/go/build.py

```diff
@@ -165,31 +130,23 @@ async def build_target(
         Digest, MergeDigests([d for _, d in import_config_digests.values()])
     )
 
-    enriched_goroot = await Get(
-        EnrichedGoLangDistribution, EnrichGoLangDistributionRequest(downloaded_goroot)
-    )
-
+    input_root_digests = {source_files.snapshot.digest, merged_packages_digest}
```

Comment:
Better to use a list than set so that this is ordered as the input to `MergeDigests`. That's required for the memoization to be consistent.

Reply:
I'm using a set here because, when we add stdlib packages, we need to deduplicate digests.

Reply:
Can I just sort the digests when converting to a list?

Reply:
You can use `OrderedSet` instead from `pants.util.ordered_set` :)